atlas/AtlasMl/atlasml/ml/MLPipelines/PipelineWorkflows.py

```
@@ -0,0 +1,139 @@
+
+        for competency in competencies:
+            uuid, embedding = generate_embeddings_local(competency["properties"]["uuid"], competency["properties"]["text"])
+            similarity_score = np.array(compute_cosine_similarity(embedding, medoid) for medoid in medoids)
+            best_medoid_idx = int(np.argmax(similarity_score))
+            properties = { "properties": [{
+                "competency_id": uuid ,
+                "name" : competency["properties"]["name"],
+                "text": competency["properties"]["text"],
+                "competencyID": clusters[best_medoid_idx]["properties"]["uuid"]
+            }]}
+            self.weaviate_client.add_embeddings(CollectionNames.COMPETENCY.value, embedding, properties)
```

Pylint (3.3.7)

[convention] Line too long (104/100)

(C0301)